pkg/k8shandler/kibana/reconciler.go

```diff
@@ -0,0 +1,635 @@
+				if errors.IsNotFound(err) {
+					// the object doesn't exist -- it was likely culled
+					// recreate it on the next time through if necessary
+					logrus.Debugf("Returning nil. The deployment %q was not found even though create previously failed.  Was it culled?", kibanaDeployment.Name)
+					return nil
+				}
+				return fmt.Errorf("Failed to get Kibana deployment: %v", err)
+			}
+
+			current, different := isDeploymentDifferent(current, kibanaDeployment)
+
+			if current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] != kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] {
+				different = true
+			}
+
+			for _, secretName := range []string{"kibana", "kibana-proxy"} {
+				hashKey := fmt.Sprintf("%s%s", constants.SecretHashPrefix, secretName)
+				if kibanaDeployment.Spec.Template.ObjectMeta.Annotations[hashKey] != current.Spec.Template.ObjectMeta.Annotations[hashKey] {
+					different = true
+				}
+			}
+
+			if different {
+				current.Spec = kibanaDeployment.Spec
+				return clusterRequest.Update(current)
+			}
+			return nil
+		})
+	}
+
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) getKibanaAnnotations(deployment *apps.Deployment) (map[string]string, error) {
+
+	if deployment.Spec.Template.ObjectMeta.Annotations == nil {
+		deployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+	}
+
+	annotations := deployment.Spec.Template.ObjectMeta.Annotations
+
+	kibanaTrustBundle := &v1.ConfigMap{}
+	kibanaTrustBundleName := types.NamespacedName{Name: constants.KibanaTrustedCAName, Namespace: constants.OpenshiftNS}
+	if err := clusterRequest.client.Get(context.TODO(), kibanaTrustBundleName, kibanaTrustBundle); err != nil {
+		if !errors.IsNotFound(err) {
+			return annotations, err
+		}
+	}
+
+	if _, ok := kibanaTrustBundle.Data[constants.TrustedCABundleKey]; !ok {
+		return annotations, fmt.Errorf("%v does not yet contain expected key %v", kibanaTrustBundle.Name, constants.TrustedCABundleKey)
+	}
+
+	trustedCAHashValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
+	if err != nil {
+		return annotations, fmt.Errorf("unable to calculate trusted CA value. E: %s", err.Error())
+	}
+
+	if trustedCAHashValue == "" {
+		return annotations, fmt.Errorf("Did not receive hashvalue for trusted CA value")
+	}
+
+	annotations[constants.TrustedCABundleHashName] = trustedCAHashValue
+
+	//generate secret hash
+	for _, secretName := range []string{"kibana", "kibana-proxy"} {
+
+		hashKey := fmt.Sprintf("%s%s", constants.SecretHashPrefix, secretName)
+
+		secret, err := clusterRequest.GetSecret(secretName)
+		if err != nil {
+			return annotations, err
+		}
+		secretHashValue, err := calcSecretHashValue(secret)
+		if err != nil {
+			return annotations, err
+		}
+
+		annotations[hashKey] = secretHashValue
+	}
+
+	return annotations, nil
+}
+
+func (clusterRequest *KibanaRequest) GetSecret(secretName string) (*core.Secret, error) {
```

Comment:
non-blocking nit but this should probably be in the `kibana/secret.go` -- we can do this as tech-debt